### PR TITLE
Add transform namespace

### PR DESCRIPTION
**Pull request by karstendick**
Adds a `transform` function to coerce data into the types specified in the schema.

### CODE REVIEW

File: singer/transform.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            data = data.replace(',', '')
+        return float(data)
+
+    elif typ == "boolean":
+        return bool(data)
+
+    else:
+        raise Exception("Invalid type: {}".format(typ))
+
+def transform(data, schema, integer_datetime_fmt=NO_INTEGER_DATETIME_PARSING):
```
Comment (mdelaurentis) [inline singer/transform.py]: A docstring here would be good

File: singer/transform.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    else:
+        try:
+            if integer_datetime_fmt == UNIX_SECONDS_INTEGER_DATETIME_PARSING:
+                return unix_seconds_to_datetime(value)
+            elif integer_datetime_fmt == UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING:
+                return unix_milliseconds_to_datetime(value)
+        except:
+            return string_to_datetime(value)
+
+NO_INTEGER_DATETIME_PARSING = 0
```
Comment (mdelaurentis) [inline singer/transform.py]: I think I'd prefer string constants rather than ints. Is there a reason why we couldn't use string constants here?

Comment (karstendick) [inline singer/transform.py]: Not at all. I really wanted to use a Clojure keyword here ...

File: singer/transform.py (hunk 1 above)
Comment (karstendick) [inline singer/transform.py]: ✔️

**Pull request merged**